cluster-autoscaler/processors/README.md

```diff
@@ -0,0 +1,32 @@
+# Processor interfaces
+
+Processors are golang interfaces that allow modifying key Cluster Autoscaler
+data structures at various points in CA logic. This makes them a powerful tool
+for building advanced features and customizing CA behavior.
+
+  * Processors are intended to simplify management of customized CA fork by
+    allowing most (and hopefully all) custom logic to be encapsulated behind a
+    few interfaces.
+  * Processors implementing generic features that may be useful to a large
+    number of CA users may be merged to kubernetes/autoscaler repository.
+
+Processors are the *preferred way* for introducing new features to core CA
+logic. If a feature can be implemented via an existing processor interface, the
+reviewers may ask for a PR to be implemented this way.
+
+## Changes and deprecations
+
+Adding new processors, adding new methods to existing processors or adding
+additional parameters to existing methods may all happen in minor releases of
+Cluster Autoscaler.
```

Comment:
Open question: should we also name the cloudprovider interface here (ie. can that interface follow the same evolution process - and if not, should this FAQ document it)?

Reply:
That is a very good question. I think it is how we've been treating cloudprovider historically (at least the addition part - ex. introduction of GetOptions). It could also apply to other interfaces we expect people to commonly implement in forks (expander?).

@mwielgus @gjtempleton Any opinions on this one? I'm happy to move this to a top-level deprecation policy doc that covers all of the above.